java/src/main/native/include/jni_utils.hpp

```diff
@@ -790,19 +756,42 @@ inline void jni_cuda_check(JNIEnv *const env, cudaError_t cuda_status) {
     JNI_THROW_NEW(env, class_name, message, ret_val)                                               \
   }
 
+// Throw a new exception only if one is not pending then always return with the specified value
+#define JNI_CHECK_THROW_NEW_CUDA_ERROR(env, class_name, e, ret_val)                                \
+  do {                                                                                             \
+    if (env->ExceptionOccurred()) {                                                                \
+      return ret_val;                                                                              \
+    }                                                                                              \
+    std::string n_msg = std::string("CUDA ERROR: ") + (e.what() == nullptr ? "" : e.what());       \
```

Comment:
`CUDA ERROR` is somewhat redundant here because the exception type we're throwing already implies it.

Reply:
Fixed.